src/views/mypage/MyPageView.vue

```diff
@@ -91,49 +102,76 @@
         @cancel="showModal = false"
         @confirm="logout"
       />
+
+      <ConfirmModal
+        v-if="showChoogoomiEditModal"
+        title="추구 유형 수정"
+        :message="'추구 유형을 수정하면\n모든 랭킹과 점수가 초기화됩니다.\n그래도 수정하시겠습니까?'"
+        :cancelBtn="'취소'"
+        :confirmBtn="'확인'"
+        @cancel="showChoogoomiEditModal = false"
+        @confirm="router.push('/character')"
+      />
     </div>
   </div>
   <BottomNavigation />
 </template>
 
 <script setup>
-import { computed, ref } from 'vue';
+import { onMounted, reactive, ref } from 'vue';
 
-import character_savings from '@/assets/img/characters/character_savings.png';
+import axiosInstance from '@/api/axios';
 import edit from '@/assets/img/icons/feature/icon_edit.png';
 import BottomNavigation from '@/components/BottomNavigation.vue';
 import ConfirmModal from '@/components/ConfirmModal.vue';
 import TopNavigation from '@/components/TopNavigation.vue';
+import { CHOOGOOMI_MAP } from '@/constants/choogoomiMap';
 import router from '@/router';
+import { isEditableDay } from '@/utils/dateUtils';
+import { getLevel, LEVEL_THRESHOLDS } from '@/utils/levelUtils';
 
 import MyPageBtn from './components/MyPageBtn.vue';
 
 const showModal = ref(false);
+const showChoogoomiEditModal = ref(false);
 
-const isEditableDay = computed(() => {
-  const today = new Date();
-  const year = today.getFullYear();
-  const month = today.getMonth();
-
-  // 해당 달의 1일부터 7일까지
-  for (let day = 1; day <= 7; day++) {
-    const date = new Date(year, month, day);
-
-    // 일요일 찾음
-    if (date.getDay() === 0) {
-      //오늘이 첫째주 일요일이라면
-      return (
-        today.getDate() === date.getDate() &&
-        today.getMonth() === date.getMonth() &&
-        today.getFullYear() === date.getFullYear()
-      );
-    }
-  }
-  return false;
+const choogoomiImage = ref('');
+const userLevel = ref(0);
+
+const userInfo = reactive({
+  choogoomiName: '',
+  nickname: '',
+  userScore: null,
+  userRanking: null,
+  isLevelUp: false,
 });
 
+const editableDay = isEditableDay();
+
 const logout = () => {
   localStorage.clear();
   router.push('/login');
 };
+
+onMounted(async () => {
+  try {
+    const { data } = await axiosInstance.get('/api/users/main-profile');
+
+    //userScore로 레벨 계산
+    userLevel.value = getLevel(data.userScore);
+
+    //추구미 알파벳을 유형명으로 매핑
+    const choogoomi = CHOOGOOMI_MAP[data.choogooMi][userLevel.value];
```

Comment:
```suggestion
const choogoomi = CHOOGOOMI_MAP.find(item => item.choogoomiName === data.choogooMi).userLevel[userLevel.value];
```
배열 형태에서는 이런 식으로 'find'를 사용해 접근하는게 더 좋다고 합니당 ㅎㅎㅎ 수정하지는 않으셔도 됩니다!

Reply:
네 감사합니다!!